.github/workflows/codegen-build-test.yml

```diff
@@ -0,0 +1,49 @@
+# This script is for building the full SDK
+# It will regenerate all models, build the full SDK with the new models, and run tests.
+name: Codegen, Build, and Test
+
+on:
+  workflow_dispatch:
+
+env:
+  BUILDER_VERSION: v0.8.19
+  BUILDER_SOURCE: releases
+  # host owned by CRT team to host aws-crt-builder releases. Contact their on-call with any issues
+  BUILDER_HOST: https://d19elf31gohf1l.cloudfront.net
+  PACKAGE_NAME: aws-sdk-swift
+  LINUX_BASE_IMAGE: ubuntu-16-x64
+  RUN: ${{ github.run_id }}-${{ github.run_number }}
+  AWS_SDK_SWIFT_CI_DIR: /Users/runner/work/aws-sdk-swift/aws-sdk-swift
+  AWS_CRT_SWIFT_CI_DIR: /Users/runner/work/aws-sdk-swift/aws-sdk-swift/target/build/deps/aws-crt-swift
+  SMITHY_SWIFT_CI_DIR: /Users/runner/work/aws-sdk-swift/aws-sdk-swift/target/build/deps/smithy-swift
+
+jobs:
+  codegen-build-test:
+    runs-on: macos-11
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '11'
+      - name: Build and Test ${{ env.PACKAGE_NAME }}
+        run: |
+          python3 -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder.pyz')"
+          chmod a+x builder.pyz
+          AWS_CRT_SWIFT_CI_DIR="${{ env.AWS_CRT_SWIFT_CI_DIR }}" AWS_SDK_SWIFT_CI_DIR="${{ env.AWS_SDK_SWIFT_CI_DIR }}" SMITHY_SWIFT_CI_DIR="${{ env.SMITHY_SWIFT_CI_DIR }}" ./builder.pyz build -p ${{ env.PACKAGE_NAME }}
+          ./gradlew -p codegen/sdk-codegen build
+          ./gradlew -p codegen/sdk-codegen stageSdks
```

Comment:
I would `rm -rf release/` before running `stageSdks` to ensure that no files from previous generation stick around for your build

Reply:
ooc why isn't that an issue in the `release.yml` workflow?

Reply:
Interesting that it's not in there, but it should be.  (It wasn't necessary back when we used to not commit `release/` since there were no pre-existing files; not adding `rm-rf release/` when we did the work to make `main` buildable was an oversight on my part.)

If you add that command in your new workflow and the existing release workflow, I will approve.